rust_dev_preview/examples/auto-scaling/src/scenario.rs

```diff
@@ -0,0 +1,447 @@
+use std::{
+    fmt::Display,
+    time::{Duration, SystemTime},
+};
+
+use anyhow::anyhow;
+use aws_sdk_autoscaling::{
+    error::DisplayErrorContext,
+    types::{Activity, AutoScalingGroup, LaunchTemplateSpecification},
+};
+use aws_sdk_ec2::types::RequestLaunchTemplateData;
+use tokio_stream::StreamExt;
+
+const LAUNCH_TEMPLATE_NAME: &str = "SDK_Code_Examples_EC2_Autoscaling_template_from_Rust_SDK";
+const AUTOSCALING_GROUP_NAME: &str = "SDK_Code_Examples_EC2_Autoscaling_Group_from_Rust_SDK";
+const MAX_WAIT: Duration = Duration::from_secs(5);
+
+pub struct AutoScalingScenario {
+    ec2: aws_sdk_ec2::Client,
+    autoscaling: aws_sdk_autoscaling::Client,
+    launch_template_arn: String,
+    auto_scaling_group_name: String,
+}
+
+impl Display for AutoScalingScenario {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        f.write_fmt(format_args!(
+            "\tLaunch Template ID: {}\n",
+            self.launch_template_arn
+        ))?;
+        f.write_fmt(format_args!(
+            "\tScaling Group Name: {}\n",
+            self.auto_scaling_group_name
+        ))?;
+
+        Ok(())
+    }
+}
+
+pub struct AutoScalingScenarioDescription {
+    instances: Result<Vec<String>, anyhow::Error>,
+    activities: Result<Vec<Activity>, anyhow::Error>,
+}
+
+impl Display for AutoScalingScenarioDescription {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        f.write_str("\t         Instances:\n")?;
+        match &self.instances {
+            Ok(instances) => {
+                for instance in instances {
+                    f.write_fmt(format_args!("\t\t- {instance}\n"))?;
+                }
+            }
+            Err(e) => f.write_fmt(format_args!("\t\t! {e}"))?,
+        }
```

Comment:
kinda weird to have both mixed tabs and spaces and whitespace in the format output 🤔 

might want to use https://doc.rust-lang.org/std/fmt/struct.Formatter.html#method.width or something to control this 🤔